sdk/src/main/scala/com/horizen/account/websocket/data/Subscription.scala

```diff
@@ -0,0 +1,49 @@
+package com.horizen.account.websocket.data
+
+import com.horizen.account.receipt.Bloom
+import com.horizen.evm.interop.EvmLog
+import com.horizen.evm.utils.Hash
+import com.horizen.utils.BytesUtils
+import org.web3j.utils.Numeric
+
+import jakarta.websocket.Session
+
+trait BaseSubscription {
+  def session: Session
+  def subscriptionId: String
+}
+
+case class Subscription(session: Session, subscriptionId: String) extends BaseSubscription {}
+case class SubscriptionWithFilter(session: Session, subscriptionId: String,
+                                  address: Option[Array[String]],
+                                  topics: Option[Array[String]]) extends BaseSubscription {
+
+  def checkSubscriptionInBloom(bloom: Bloom): Boolean = {
+    if (address.isDefined && !address.get.exists(addr => bloom.test(BytesUtils.fromHexString(Numeric.cleanHexPrefix(addr)))))
+      false
+    else
+      filterBlockLogsByTopic(bloom)
+  }
+
+  def filterBlockLogsByTopic(bloom: Bloom): Boolean = {
+    if (topics.isDefined && !topics.get.exists(topic => bloom.test(BytesUtils.fromHexString(Numeric.cleanHexPrefix(topic)))))
+      false
+    else
+      true
+  }
```

Comment:
address and topics of a specific subscription are immutable.
During the instance creation you may calculate and store the bloom filter of it.
When, someone calls `checkSubscriptionInBloom` just check one one bloom against the other. Note: a new `test/contains` method in Bloom class may require

Reply:
Changed the logic, now I reuse the **getBlockLogs** taken from the EthService class